packages/combobox/src/Combobox.types.ts

```diff
@@ -305,30 +309,46 @@ interface BaseComboboxOptionProps extends ComponentPropsWithoutRef<'li'> {
    * Styling Prop
    */
   className?: string;
+
+  /**
+   * Optional descriptive text under the displayName
+   */
+  description?: string;
+
+  /**
+   * Callback fired when an option is clicked. Returns the event and the option value.
+   */
+  onClick?: (
+    event: React.SyntheticEvent<HTMLLIElement, Event>,
+    value: string,
+  ) => void;
 }
 
+type BaseComboboxOptionProps = ListItemProps &
+  BaseOptionalProps &
+  Partial<BaseValueProps>;
```

Comment:
Partial<> indicates that those properties are optional - why are we splitting up OptionalProps and ValueProps? Can they be combined into one interface?

Reply:
I was trying to split these up since `ComboboxOptionProps` and `InternalComboboxOptionProps` have similar props but value and displayName are required in `InternalComboboxOptionProps`. One option can be to maybe make `BaseValueProps` all optional and use `Required<BaseValueProps>` inside `InternalComboboxOptionProps`. I'll see if i can improve on this.

Reply:
I initially wrote a reply to this, then deleted it cause I thought it was incorrect. `ComboboxOptionProps`, `OptionObject`, and `InternalComboboxOptionProps` share similar props but `value` and `displayName` are required in both `OptionObject` and `InternalComboboxOptionProps`. One option i can think of is to make `value` and `displayName` optional inside `BaseValueProps` but use `Required<BaseValueProps>`  inside `OptionObject`  and `InternalComboboxOptionProps`. There are prob other ways to improve this, so I'll mess around with it.